python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py

```diff
@@ -269,15 +277,31 @@ class AirbyteCloudResource(BaseAirbyteResource):
         )
     """
 
-    api_key: str = Field(..., description="The Airbyte Cloud API key.")
+    client_id: str = Field(..., description="The Airbyte Cloud client ID.")
+    client_secret: str = Field(..., description="The Airbyte Cloud client secret.")
+
+    _access_token_value: Optional[str] = PrivateAttr(default=None)
+    _access_token_timestamp: Optional[float] = PrivateAttr(default=None)
+
+    def setup_for_execution(self, context: InitResourceContext) -> None:
+        # Refresh access token when the resource is initialized
+        self._refresh_access_token()
 
     @property
     def api_base_url(self) -> str:
         return "https://api.airbyte.com/v1"
 
     @property
     def all_additional_request_params(self) -> Mapping[str, Any]:
-        return {"headers": {"Authorization": f"Bearer {self.api_key}", "User-Agent": "dagster"}}
+        # Make sure the access token is refreshed before using it when calling the API.
```

Comment:
Since this is a `@property`, we should avoid doing real work inside the fn body (like refreshing the access token). One different approach could be be to pull out the refresh into a separate fn like `_get_or_refresh_access_token()` so that it's clear that invoking it may have a cost, and call that directly in `make_request`

Reply:
Moved to `make_request` in https://github.com/dagster-io/dagster/pull/23451/commits/f2fefcb5efa4a2a7e9134baff85519a0177ab74e